### PR TITLE
Allow logging into specific subscription

### LINKED ISSUE #1404: Specifying SubscriptionId should not require Provisioner parameter set
For people outside our playground or testing subscriptions (e.g. partner teams), they may need to specify a specific subscription. Currently, the `SubscriptionId` parameter for the TestResources script is part of the provisioner parameter set, and then requires that provisioner parameters be set. This really doesn't need to be the case, as `SubscriptionId` is really only used to log in if not already logged in, which then, by default, associates resources with that subscription.

### PR DESCRIPTION
**Pull request by heaths**
Resolves #1404

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]
Looks reasonable to me.

**Review by benbp — APPROVED** [review]

**Comment by benbp** [comment]
It would be nice to not default this to playground with `$CI -eq $true`, but I'm not sure there's a great way to do that with your parameter change.

**Comment by heaths** [comment]
The overall flow won't change. `$CI` was only `$true` if `$SubscriptionId` and other provisioning parameters were specified anyway. This just refactors the value in a way that can be overridden. Before it was hardcoded (calling it a "default" wasn't true because it was always *the* value). This might also fix the problem some people outside our team have reported that throws an exception from the Az cmdlets - a `NullReferenceException`, IIRC. It seemed to be caused by an account context belonging to a different subscription than what we tried to log into. 🤞

**Comment by benbp** [comment]
Resolves #1319

File: eng/common/TestResources/New-TestResources.ps1 [hunk 1]
```diff
@@ -202,7 +201,7 @@ try {
         $context = Get-AzContext;
         if (!$context) {
             Log "You are not logged in; connecting to 'Azure SDK Developer Playground'"
```
Comment (pakrym) [inline eng/common/TestResources/New-TestResources.ps1:203]: This message is not correct anymore.

Comment (heaths) [reply eng/common/TestResources/New-TestResources.ps1:203]: Thanks. I missed that, but not really worth starting this process over. I opened https://github.com/Azure/azure-sdk-tools/issues/1407 to track. Maybe we could use a small look-up table of subscription IDs to use a friendly name when possible.

**Pull request merged**